framework/src/utils/PetscDMMoose.C

```diff
@@ -631,7 +632,7 @@ DMMooseGetEmbedding_Private(DM dm, IS * embedding)
               PenetrationInfo * pinfo = lit->second;
               std::map<dof_id_type, std::vector<dof_id_type> > & node_to_elem_map =
               dmm->_nl->_fe_problem.mesh().nodeToElemMap();
-              if (pinfo && pinfo->isCaptured())
+              if (pinfo && (pinfo->isCaptured() || dmm->_include_all_contact_nodes))
```

Comment:
This would only pick up the master nodes that are on faces that are initially paired with slave nodes. I think we need to iterate through both surfaces and add everything for this option.